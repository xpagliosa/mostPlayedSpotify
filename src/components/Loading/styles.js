import { StyleSheet } from 'react-native';
import { general, metrics, colors, fonts } from '../../styles';

const styles = StyleSheet.create({
    loadingContainer: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },

    loading: {
        color: colors.white,
        fontSize: fonts.big
    },

    button: {
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 3,
        backgroundColor: colors.light,
        marginTop: metrics.smallMargin,
        padding: metrics.smallMargin,
      },
    
      buttonText: {
        fontWeight: 'bold',
        color: colors.darker,
        fontSize: fonts.regular,
      },
      
})

export default styles;