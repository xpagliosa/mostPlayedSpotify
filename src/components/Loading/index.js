import React, { Component } from 'react';

import { ActivityIndicator, Text, TouchableOpacity, View  } from 'react-native';

import styles from './styles';

export default class Loading extends Component {

  render(){
    return(
      <View style={styles.loadingContainer}>
        <ActivityIndicator size="large" color="#FFF" />
        <Text style={styles.loading}> {this.props.loadingText} </Text>
        { this.props.tryAgain && 
          <TouchableOpacity
            style={styles.button}
            onPress={this.props.signIn}
          >
            <Text style={styles.buttonText}>Tentar Novamente</Text>
          </TouchableOpacity>
        }
      </View>
    );
  }
}
