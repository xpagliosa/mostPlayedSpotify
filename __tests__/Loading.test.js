import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import Loading from '../src/components/Loading';

describe('Snapshot Loading', () => {
    let wrapper = shallow(
        <Loading 
            loadingText="Conectando Spotify..."
            tryAgain={false}
        />
    );
    it('shouls render Loading correctly', () => {
        expect(wrapper).toMatchSnapshot();
    }),
    it('shouls render tryAgain correctly', () => {
        wrapper.setProps({ tryAgain: true });
        expect(wrapper).toMatchSnapshot();
    })
});

// describe('rendering Loading', () => {
//     let wrapper
//     beforeEach(() => {
//         wrapper = shallow(
//             <Loading 
//                 loadingText="Conectando Spotify..."
//                 tryAgain={false}
//             />
//         );
//     })
//     describe('<View />', () => {
//         it('should reander a <View />', () => {
//             expect(wrapper.find('View')).toHaveLength(1)
//         })
//     })
//     describe('<ActivityIndicator />', () => {
//         it('should render a <ActivityIndicator />', () => {
//             expect(wrapper.find('ActivityIndicator')).toHaveLength(1)
//         })
//     })
    
//     describe('<Text />', () => {
//         it('should render a Loading Text', () => {
//             expect(wrapper.find('Text').contains('Conectando Spotify...')).toBe(true)
//         })
//     })

//     describe('NO tryAgain Button', () => {
//         it('should render a <TouchableOpacity />', () => {
//             expect(wrapper.find('TouchableOpacity')).toHaveLength(0)
//         })
//     })
    
//     describe('YES tryAgain Button', () => {
//         beforeEach(() => {
//             wrapper = shallow(
//                 <Loading 
//                     loadingText="Conectando Spotify..."
//                     tryAgain={true}
//                 />
//             );
//         })

//         it('should render a <TouchableOpacity />', () => {
//             expect(wrapper.find('TouchableOpacity')).toHaveLength(1)
//         })
//         it('should render a Label', () => {
//             expect(wrapper.find('Text').contains('Tentar Novamente')).toBe(true)
//         })
//     })
    
// });