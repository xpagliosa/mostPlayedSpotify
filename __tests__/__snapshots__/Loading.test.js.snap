// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Loading shouls render Loading correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading
    loadingText="Conectando Spotify..."
    tryAgain={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ActivityIndicator
          animating={true}
          color="#FFF"
          hidesWhenStopped={true}
          size="large"
        />,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFF",
              "fontSize": 18,
            }
          }
        >
           
          Conectando Spotify...
           
        </Text>,
        false,
      ],
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animating": true,
          "color": "#FFF",
          "hidesWhenStopped": true,
          "size": "large",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": Array [
            " ",
            "Conectando Spotify...",
            " ",
          ],
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#FFF",
            "fontSize": 18,
          },
        },
        "ref": null,
        "rendered": Array [
          " ",
          "Conectando Spotify...",
          " ",
        ],
        "type": [Function],
      },
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ActivityIndicator
            animating={true}
            color="#FFF"
            hidesWhenStopped={true}
            size="large"
          />,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFF",
                "fontSize": 18,
              }
            }
          >
             
            Conectando Spotify...
             
          </Text>,
          false,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animating": true,
            "color": "#FFF",
            "hidesWhenStopped": true,
            "size": "large",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              " ",
              "Conectando Spotify...",
              " ",
            ],
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#FFF",
              "fontSize": 18,
            },
          },
          "ref": null,
          "rendered": Array [
            " ",
            "Conectando Spotify...",
            " ",
          ],
          "type": [Function],
        },
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Snapshot Loading shouls render tryAgain correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading
    loadingText="Conectando Spotify..."
    tryAgain={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ActivityIndicator
          animating={true}
          color="#FFF"
          hidesWhenStopped={true}
          size="large"
        />,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFF",
              "fontSize": 18,
            }
          }
        >
           
          Conectando Spotify...
           
        </Text>,
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={undefined}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#DDD",
              "borderRadius": 3,
              "justifyContent": "center",
              "marginTop": 10,
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            Tentar Novamente
          </Text>
        </TouchableOpacity>,
      ],
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animating": true,
          "color": "#FFF",
          "hidesWhenStopped": true,
          "size": "large",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": Array [
            " ",
            "Conectando Spotify...",
            " ",
          ],
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#FFF",
            "fontSize": 18,
          },
        },
        "ref": null,
        "rendered": Array [
          " ",
          "Conectando Spotify...",
          " ",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            Tentar Novamente
          </Text>,
          "onPress": undefined,
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "#DDD",
            "borderRadius": 3,
            "justifyContent": "center",
            "marginTop": 10,
            "padding": 10,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Tentar Novamente",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#333",
              "fontSize": 14,
              "fontWeight": "bold",
            },
          },
          "ref": null,
          "rendered": "Tentar Novamente",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ActivityIndicator
            animating={true}
            color="#FFF"
            hidesWhenStopped={true}
            size="large"
          />,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFF",
                "fontSize": 18,
              }
            }
          >
             
            Conectando Spotify...
             
          </Text>,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={undefined}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#DDD",
                "borderRadius": 3,
                "justifyContent": "center",
                "marginTop": 10,
                "padding": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#333",
                  "fontSize": 14,
                  "fontWeight": "bold",
                }
              }
            >
              Tentar Novamente
            </Text>
          </TouchableOpacity>,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animating": true,
            "color": "#FFF",
            "hidesWhenStopped": true,
            "size": "large",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              " ",
              "Conectando Spotify...",
              " ",
            ],
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#FFF",
              "fontSize": 18,
            },
          },
          "ref": null,
          "rendered": Array [
            " ",
            "Conectando Spotify...",
            " ",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#333",
                  "fontSize": 14,
                  "fontWeight": "bold",
                }
              }
            >
              Tentar Novamente
            </Text>,
            "onPress": undefined,
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "#DDD",
              "borderRadius": 3,
              "justifyContent": "center",
              "marginTop": 10,
              "padding": 10,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Tentar Novamente",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#333",
                "fontSize": 14,
                "fontWeight": "bold",
              },
            },
            "ref": null,
            "rendered": "Tentar Novamente",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
